### DISCUSSION TITLE
how to abort build that is stuck in "no suitable workers found" step

### DISCUSSION THREAD

**Discussion by alex1x** [concourse/concourse#6955] [Help & Support]
Hi, 

I have a few builds that are stuck in the "no suitable workers found, waiting for worker..." step. I am using "limit-active-tasks" placement strategy. I know that my builds will never recover because there will never be the "suitable" workers they are expecting. However these builds have been stuck waiting for a suitable worker for almost 24 hours now. Aborting either via the UI or the CLI doesn't work. Concourse Version is 7.2.

What can I do about these builds?

Thanks,
Alex

**Comment by taylorsilva** [comment]
Sorry for the late reply! This sounds like a bug; you should be able to abort a build at any point.

To get around this, the easiest thing I can think of doing would be to restart all your web nodes. The build is being tracked/managed from one of the web nodes. Restarting the web node should cause the build to fail.

**Reply by taylorsilva** [reply]
I tried reproducing this issue locally but could not. If you're able to reproduce this and submit it as an [issue](https://github.com/concourse/concourse/issues) it would be greatly appreciated!

**Reply by taylorsilva** [reply]
Might be similar to #6762

**Reply by alex1x** [reply]
Thanks! The analysis in #6762 makes sense to me and I think it's a similar issue. Here are the log messages I found for this build ID on my end:

```
{
    "timestamp": "2021-05-04T16:32:52.544322491Z",
    "message": "atc.tracker.tick.run.releasing",
},
{
    "timestamp": "2021-05-04T16:32:52.544322491Z",
    "message": "atc.tracker.tick.run.releasing",
},
{
    "timestamp": "2021-05-04T16:32:52.544322491Z",
    "message": "atc.tracker.tick.run.releasing",
},
{
    "timestamp": "2021-05-04T16:35:27.280131398Z",
    "message": "atc.tracker.tick.run.lock.acquire.acquired",
},
{
    "timestamp": "2021-05-04T16:35:27.314200169Z",
    "message": "atc.tracker.tick.run.running",
},
{
    "timestamp": "2021-05-04T16:35:27.319319247Z",
    "message": "atc.tracker.tick.run.aborting",
},
{
    "timestamp": "2021-05-04T16:35:27.366978807Z",
    "message": "atc.tracker.tick.run.get-step.initializing",
},
{
    "timestamp": "2021-05-04T16:35:27.612792994Z",
    "message": "atc.tracker.tick.run.get-step.found-existing-container-on-worker",
},
{
    "timestamp": "2021-05-04T16:35:27.735787589Z",
    "message": "atc.tracker.tick.run.get-step.starting",
},
{
    "timestamp": "2021-05-04T16:35:28.091606964Z",
    "message": "atc.tracker.tick.run.get-step.found-initialized-versioned-source",
},
{
    "timestamp": "2021-05-04T16:35:28.391958892Z",
    "message": "atc.tracker.tick.run.get-step.finished",
},
{
    "timestamp": "2021-05-04T16:35:28.392005196Z",
    "message": "atc.tracker.tick.run.get-step.attempted-to-wake-waiting-step",
},
{
    "timestamp": "2021-05-04T16:35:28.439939590Z",
    "message": "atc.tracker.tick.run.set-pipeline-step.initializing",
},
{
    "timestamp": "2021-05-04T16:35:28.837141030Z",
    "message": "atc.tracker.tick.run.errored",
}
```

I also found some messages about:

```
Put "/volumes/b2aa3e21-f045-4480-4bbb-23ab21b688bb/stream-out?path=pipeline.yml": context canceled
```

I also think that restarting the web nodes will make them go away, I will write back if it doesn't.

**Reply by alex1x** [reply]
After restarting the web node (in fact terminating it and replacing it with a new node) the job is still running. As a next step I was going to try and delete via the DB but let me know if you have a better suggestion.

**Reply by taylorsilva** [reply]
Was afraid that would happen. The build tracker keeps picking it up. How interesting that it doesn't see the abort signal even after pick it back up.

If you find the build in the `builds` table can tell us:
- what's the value of the `aborted` column?
- what's the value of the `status` column?

If the `aborted` column has a value of `false`, try changing it to `true`. The next time you restart the web nodes, the build tracker _should_ see it's aborted and not try tracking it at all.

**Reply by alex1x** [reply]
Hi, after setting aborted to `t` and restarting the web nothing changed. `status` was `started`. 

I've resolved this now by creating the "suitable worker" it was expecting... BTW just one more thing, the "stuck" part was the slack part, while the `set_pipeline` part did correctly get interrupted 2 weeks ago when I pressed the X. I wonder if that changes your assessment of the situation.

<img width="1652" alt="Screenshot 2021-05-14 at 14 40 38" src="https://user-images.githubusercontent.com/6977670/118278997-856b7800-b4c2-11eb-8b9d-fa08aa35b98c.png">

**Reply by taylorsilva** [reply]
So it was stuck trying to execute the `on_error` step? Is that on the entire job or that single task step that never even ran?

**Reply by alex1x** [reply]
It was stuck trying to schedule the `on_error` step. The `set_pipeline` was interrupted, but the overall build remained yellow and the `on_error` step was trying to schedule on a worker it couldn't find ("no suitable workers found"). Eventually I gave it the worker it wanted and the `on_error` scheduled, it posted the slack notification, and then the entire build finished (from yellow to brown color).
